go.mod

```diff
@@ -62,11 +61,11 @@ require (
 	github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca // indirect
 	github.com/zmap/zcrypto v0.0.0-20200911161511-43ff0ea04f21 // indirect
 	github.com/zmap/zlint/v2 v2.2.1 // indirect
-	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 // indirect
+	golang.org/x/crypto v0.0.0-20220214200702-86341886e292 // indirect
 	golang.org/x/exp v0.0.0-20200224162631-6cc2880d07d6 // indirect
-	golang.org/x/mod v0.4.2 // indirect
-	golang.org/x/net v0.0.0-20211209124913-491a49abca63 // indirect
-	golang.org/x/sys v0.0.0-20211029165221-6e7872819dc8 // indirect
+	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 // indirect
+	golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd // indirect
+	golang.org/x/sys v0.0.0-20220209214540-3681064d5158 // indirect
```

Comment:
Because of the breaking change on controller-runtime, I think we need to do here:

- add  set LeaderElectionResourceLock to "configmapsleases"  so that we do not change the behaviour for the default scaffolds
- ensure that it has a comment on top explaining that default behaviour change in 0.12 and has a link for the users to check the steps in: https://github.com/kubernetes-sigs/controller-runtime/pull/1903/files or we copy them.
- also, we need a new issue in the repo tracked under the go/v4 milestone to remove this option for go/v4 and we will need to doc the required changes in the migration guide from go/v3 to go/v4 

c/c @rashmigottipati @ryantking @varshaprasad96